src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -64,7 +65,9 @@ public function setContainer(ContainerInterface $container)
     protected function getParameter(string $name)
     {
         if (!$this->container->has('parameter_bag')) {
-            throw new \LogicException('The "parameter_bag" service is not available. Try running "composer require dependency-injection:^4.1"');
+            throw new ServiceNotFoundException('parameter_bag', null, null, array(),
+                'The "parameter_bag" service could not be located. Ensure that symfony/dependency-injection 4.1.0 or higher '.
```

Comment:
it would be great to decide ourselves instead of this "or" in the message (a class_exists check and we can know for sure which alternative is the correct one)

Reply:
Any ideas on a class to reliably test that was introduced in 4.1 and will not disappear until "infinity", so not internal?

Reply:
ContainerBagInterface :)

Reply:
Done.